draft-plugin/server.go

```diff
@@ -0,0 +1,200 @@
+	email          string
+	gc             git.ClientFactory
+	ghc            github.Client
+	log            *logrus.Entry
+	repos          []github.Repo
+	prowconfig     string
+	prowjob        string
+	regex          string
+	ns             string
+}
+
+type githubClient interface {
+	GetRef(org, repo, ref string) (string, error)
+}
+
+type prowJobClient interface {
+	Create(context.Context, *prowapi.ProwJob, metav1.CreateOptions) (*prowapi.ProwJob, error)
+	List(ctx context.Context, opts metav1.ListOptions) (*prowapi.ProwJobList, error)
+	Update(context.Context, *prowapi.ProwJob, metav1.UpdateOptions) (*prowapi.ProwJob, error)
+}
+
+func (s *server) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	logrus.Info("inside http server")
+	eventType, eventGUID, payload, ok, _ := github.ValidateWebhook(w, r, s.tokenGenerator)
+	if !ok {
+		return
+	}
+	logrus.Info(w, "Event received. Have a nice day.")
+	if err := s.handleEvent(eventType, eventGUID, payload); err != nil {
+		logrus.WithError(err).Error("Error parsing event.")
+	}
+}
+
+func (s *server) handleEvent(eventType, eventGUID string, payload []byte) error {
+	l := logrus.WithFields(
+		logrus.Fields{
+			"event-type":     eventType,
+			github.EventGUID: eventGUID,
+		},
+	)
+	_ = l
+	var c *config.Config
+	var p = getProwClient()
+	c, err := config.Load(s.prowconfig, s.prowjob, nil, "")
+	if err != nil {
+		logrus.Info(err)
+	}
+	var pr github.PullRequestEvent
+	if err := json.Unmarshal(payload, &pr); err != nil {
+		return err
+	}
+	org, repo, author, ref := orgRepoAuthor(pr.PullRequest)
+	baseSHA := ""
+	baseSHAGetter := func() (string, error) {
+		var err error
+		baseSHA, err = githubClient.GetRef(s.ghc, org, repo, "heads/"+ref)
+		if err != nil {
+			return "", fmt.Errorf("failed to get baseSHA: %w", err)
+		}
+		return baseSHA, nil
+	}
+	headSHAGetter := func() (string, error) {
+		return pr.PullRequest.Head.SHA, nil
+	}
+	presubmits, err := c.GetPresubmits(s.gc, org+"/"+repo, baseSHAGetter, headSHAGetter)
+	if err != nil {
+		fmt.Printf("server not responding %s", err.Error())
+	}
+	if len(presubmits) == 0 {
+		return nil
+	}
+	trustedResponse, err := trigger.TrustedUser(s.ghc, true, pr.Repo.Owner.Login, author, org, repo)
+	member := trustedResponse.IsTrusted
+	if err != nil {
+		return fmt.Errorf("could not check membership: %s", err)
+	}
+	if member {
+		switch pr.Action {
+		case github.PullRequestActionOpened, github.PullRequestActionSynchronize, github.PullRequestActionConvertedToDraft:
+			infomation := "{\"action\": \"" + string(pr.Action) + "\",\"author\": \"" + author + "\",\"pr\": " + strconv.Itoa(pr.Number) + "\",\"repo\": \"" + pr.Repo.Name + "\",\"SHA\": \"" + *pr.PullRequest.MergeSHA + "\"}"
+			logrus.Info(infomation)
+			if pr.PullRequest.Draft {
+				var jobs []config.Presubmit
+				for _, job := range presubmits {
+					match, err := regexp.MatchString(s.regex, job.Name)
+					if err != nil || !match {
+						continue
+					}
+					jobs = append(jobs, job)
```

Comment:
We are not taking any action on err here means whenever err comes in the `match=false` will be there so it could be simplified as follows:
```suggestion
					match, _ := regexp.MatchString(s.regex, job.Name)
					if match {
						jobs = append(jobs, job)
					}
```